### ISSUE TITLE
Insert or update on error with Async

### ISSUE THREAD

**Issue by IslandUsurper** [Mozu/mozu-php-sdk#19]
I can't tell if this is a bug or user error, but I'm going to lean towards user error since I only learned about promises in order to use the Async API.

As my app is exporting data into Mozu, I thought it would be a good idea that if something failed to insert because it already existed, it should automatically update. While it _does_ that, it doesn't return the values I expect.

Here is the code I'm using: [snippet](https://gist.github.com/IslandUsurper/69997c5038f658c86d3d)

If all of the `$products` exist in Mozu already, `$results` is an array full of `NULL`s. The update calls still happen, but the return values aren't propagated. Not that big of a deal, but it does throw reporting off. I wonder if it's because of the closure doing the updateProductVariations call that doesn't do anything with that result.

**Comment by IslandUsurper** [comment]
This has turned into a bigger problem than I first said. I'm also inserting or updating attributes, but then trying to add them to product types and products later on. I can work around it by GETing all of the attributes again after I've updated them, but it would be nice not to.

After spending time with XDebug, I found out that the MozuResult returned by the addAttributeAsync() and updateAttributeAsync() methods has a body property that is a stream, which is not the way that getAttributesAsync() worked. The documentation says I should get an AttributeCollection, which has an `items` property, and it does. But for POST methods, it doesn't, possibly because they are still encoded in the response body stream.
